sage_codes/Safe Curve code

```diff
@@ -0,0 +1,385 @@
+/// This section shows all of the proofs from their 
+/// written in line with the safecurves document.
+
+/// Here is the outline of the sage code for 
+/// the compuatation of values for proving 
+/// all safe curve criteria.
+/// Taken from https://safecurves.cr.yp.to/ (July, 2019)
+
+import os
+import sys
+
+def readfile(fn):
+  fd = open(fn,'r')
+  r = fd.read()
+  fd.close()
+  return r
+
+def writefile(fn,s):
+  fd = open(fn,'w')
+  fd.write(s)
+  fd.close()
+
+def expand2(n):
+  s = ""
+  
+  while n != 0:
+    j = 16
+    while 2**j < abs(n): j += 1
+    if 2**j - abs(n) > abs(n) - 2**(j-1): j -= 1
+  
+    if abs(abs(n) - 2**j) > 2**(j - 10):
+      if n > 0:
+        if s != "": s += " + "
+        s += str(n)
+      else:
+        s += " - " + str(-n)
+      n = 0
+    elif n > 0:
+      if s != "": s += " + "
+      s += "2^" + str(j)
+      n -= 2**j
+    else:
+      s += " - 2^" + str(j)
+      n += 2**j
+  
+  return s
+
+def requirement(fn,istrue):
+  writefile(fn,str(istrue) + '\n')
+  return istrue
+
+def verify():
```

Comment:
Just an idea, maybe add some comments on each function def to explain what it does will be good.

Reply:
Okay, I'll add them now